internal/resource-request-operator/broadcaster.go

```diff
@@ -101,58 +124,100 @@ func (b *Broadcaster) onUpdate(oldObj, newObj interface{}) {
+	currentResources := b.readNodeResources()
 	if node.Status.Phase == corev1.NodeRunning {
 		subResources(currentResources, *toDelete)
-		if err := b.writeResources(currentResources); err != nil {
-			klog.Errorf("OnAdd error: unable to write allocatable of Node %s: %s", node.Name, err)
+		b.writeNodesResources(currentResources)
+	}
+}
+
+func (b *Broadcaster) onPodAdd(obj interface{}) {
+	pod := obj.(*corev1.Pod)
+	klog.V(2).Infof("OnPodAdd: Add for pod %s\n", pod.Name)
+	if clusterID := pod.Labels[forge.LiqoOriginClusterID]; clusterID != "" {
+		klog.V(3).Infof("OnPodAdd: Pod %s passed ClusterID check\n", pod.Name)
+		toAdd := extractPodResources(pod)
+		currentResources := b.readPodResources(clusterID)
+		addResources(currentResources, toAdd)
+		if err := b.writePodsResources(clusterID, currentResources); err != nil {
+			klog.Errorf("OnPodAdd error: unable to write resources of Pod %s: %s", pod.Name, err)
 		}
+		return
 	}
+	klog.Warningf("OnPodAdd: missing ClusterID label on %s Pod. Resources of this pod will be ignored", pod.Name)
 }
 
-// write cluster resources in thread safe mode.
-func (b *Broadcaster) writeResources(newResources corev1.ResourceList) error {
-	b.offerMutex.Lock()
-	defer b.offerMutex.Unlock()
-	if newResources != nil {
-		b.allocatable = newResources.DeepCopy()
-		return nil
+func (b *Broadcaster) onPodDelete(obj interface{}) {
+	pod := obj.(*corev1.Pod)
+	klog.V(2).Infof("OnPodDelete: Delete for pod %s\n", pod.Name)
+	if clusterID := pod.Labels[forge.LiqoOriginClusterID]; clusterID != "" {
+		klog.V(3).Infof("OnPodDelete: Pod %s passed ClusterID check\n", pod.Name)
+		toAdd := extractPodResources(pod)
+		currentResources := b.readPodResources(clusterID)
+		subResources(currentResources, toAdd)
+		if err := b.writePodsResources(clusterID, currentResources); err != nil {
+			klog.Errorf("OnPodDelete error: unable to write resources of Pod %s: %s", pod.Name, err)
+		}
+		return
 	}
+	klog.Warningf("OnPodAdd: missing ClusterID label on %s Pod. Resources of this pod will be ignored", pod.Name)
+}
 
-	return fmt.Errorf("some error occurred during cluster resources read. Attempting writing nil resources")
+// write nodes resources in thread safe mode.
+func (b *Broadcaster) writeNodesResources(newResources corev1.ResourceList) {
+	b.nodeMutex.Lock()
+	defer b.nodeMutex.Unlock()
+	b.allocatable = newResources.DeepCopy()
 }
 
-// ReadResources return in thread safe mode a scaled value of the resources.
-func (b *Broadcaster) ReadResources() (corev1.ResourceList, error) {
-	b.offerMutex.RLock()
-	defer b.offerMutex.RUnlock()
-	if b.allocatable == nil {
-		return nil, fmt.Errorf("error getting cluster resources")
+// write pods resources in thread safe mode.
+func (b *Broadcaster) writePodsResources(clusterID string, newResources corev1.ResourceList) error {
+	b.podMutex.Lock()
+	defer b.podMutex.Unlock()
+	if clusterID != "" {
+		b.resourcePodMap[clusterID] = newResources.DeepCopy()
+		return nil
 	}
```

Comment:
I would personally avoid making this check, as you are already ensuring this condition cannot happen if you land here. @palexster WDYT?